test/unit/test_ranges.py

```diff
@@ -0,0 +1,223 @@
+import base64
+import json
+import os
+import os.path
+import pickle
+import shutil
+import tempfile
+import time
+import unittest
+
+from contextlib import closing
+from gzip import GzipFile
+from httplib import HTTPConnection
+
+import swift_scality_backend.server
+import swift.proxy.server
+import swift.account.server
+import swift.container.server
+
+from eventlet import listen, spawn, wsgi
+from swift.common.ring import RingData
+from swift.common.utils import normalize_timestamp
+
+
+class TestRangeHeaders(unittest.TestCase):
+    def setUp(self):
+        # Bind proxy, account, container, and object server to a free port
+        bind_address = ('127.0.0.1', 0)
+
+        # Directory holding account, container, and object rings
+        self.swift_dir = tempfile.mkdtemp()
+
+        # Setup mocked sproxyd server
+        sproxyd_socket = listen(bind_address)
+        spawn(wsgi.server, sproxyd_socket, self.sproxyd_mock)
```

Comment:
This thread is not killed at the end of a test. Consider using the `test.unit.utils.WSGIServer` context manager instead?